src/plugins/dashboard_embeddable_container/public/actions/replace_panel_flyout.tsx

```diff
@@ -76,7 +76,7 @@ export class ReplacePanelFlyout extends React.Component<Props> {
     // add the new view
     const newObj = await this.props.container.addSavedObjectEmbeddable(type, id);
 
-    const finalPanels = this.props.container.getInput().panels;
+    const finalPanels = _.cloneDeep(this.props.container.getInput().panels);
```

Comment:
What changed here, that made you clone the panels?

Reply:
State containers deeply freeze the object in dev mode, which allows to find places where we unintentionally mutate objects passed into services. The freezing pointed me to this `replace_panel_flyout`:
1. Tweaked the code to not directly mutate original panels
2. Fixed the call `container.updateInput(finalPanels)` -> `container.updateInput({panels: finalPanels)`. Typescript didn't complain because updates